iml-agent/src/action_plugins/check_ha.rs

```diff
@@ -71,27 +92,208 @@ impl AgentInfo {
+                            .map(|p| AgentInfo::create(p, None)),
+                    )
+                    .collect()),
             }
         })
-        .from_err()
+}
+
+pub fn get_ha_resource_list(_: ()) -> impl Future<Item = Vec<AgentInfo>, Error = ImlAgentError> {
+    get_ha_resource_list_async(()).boxed().compat()
+}
+/// Output of buctl to just then string:
+/// b "enabled"  ->  enabled
+fn busctl_to_string(output: &str) -> &str {
+    output.split("\"").nth(1).unwrap()
+}
+
+async fn systemd_unit_enabled(name: &str) -> Result<bool, ImlAgentError> {
+    let s = format!("{}.service", name);
+    cmd_output_success(
+        "busctl",
+        &[
+            "--system",
+            "--no-pager",
+            "call",
+            "org.freedesktop.systemd1",
+            "/org/freedesktop/systemd1",
+            "org.freedesktop.systemd1.Manager",
+            "GetUnitFileState",
+            "s",
+            &s,
+        ],
+    )
+    .compat()
+    .await
+    .map(
+        |output| match busctl_to_string(str::from_utf8(&output.stdout).unwrap()) {
+            "enabled" => true,
+            _ => false,
+        },
+    )
+}
+
+/// property: org.freedesktop.systemd1
+/// format!("/org/freedesktop/systemd1/unit/{}_2eservice", name)
+/// UnitFileState   +   ActiveState  -> RunState
+/// disabled            inactive        Stopped
+/// enabled             inactive        Enabled
+/// disabled            active          Started
+/// enabled             active          Setup
+async fn systemd_unit_props(name: &str) -> Result<RunState, ImlAgentError> {
+    let s = format!("/org/freedesktop/systemd1/unit/{}_2eservice", name);
+    cmd_output(
+        "busctl",
+        &[
+            "--system",
+            "--no-pager",
+            "get-property",
+            "org.freedesktop.systemd1",
+            &s,
+            "org.freedesktop.systemd1.Unit",
+            "UnitFileState",
+            "ActiveState",
+        ],
+    )
+    .compat()
+    .await
+    .and_then(|output| {
+        let res: Vec<&str> = str::from_utf8(&output.stdout)
+            .unwrap()
+            .lines()
+            .map(|o| busctl_to_string(o))
+            .collect();
+        match res[..] {
+            ["disabled", "inactive"] => Ok(RunState::Stopped),
+            ["enabled", "inactive"] => Ok(RunState::Enabled),
+            ["disabled", "active"] => Ok(RunState::Started),
+            ["enabled", "active"] => Ok(RunState::Setup),
+            _ => Err(ImlAgentError::from(RequiredError(format!(
+                "Unknown busctl ({}) output: {:?}",
+                s, output.stdout
+            )))),
+        }
+    })
+}
+
+async fn systemd_unit_servicestate(name: &str) -> Result<ServiceState, ImlAgentError> {
+    if systemd_unit_enabled(name).await? {
+        Ok(ServiceState::Configured(systemd_unit_props(name).await?))
+    } else {
+        Ok(ServiceState::Unconfigured)
+    }
+}
+
+async fn file_exists(path: &str) -> bool {
```

Comment:
Why not use: https://docs.rs/tokio/0.1.22/tokio/fs/fn.metadata.html

Reply:
I got errors, something about running in tokio threads. (I don't remember exactly)

Reply:
ah, right, needs to be inside a runtime. Let's wait till 0.2